src/Microsoft.Extensions.CommandLineUtils/CommandLine/CommandLineApplication.cs

```diff
@@ -313,10 +330,20 @@ public void ShowHint()
         // Show full help
         public void ShowHelp(string commandName = null)
         {
-            var headerBuilder = new StringBuilder("Usage:");
             for (var cmd = this; cmd != null; cmd = cmd.Parent)
             {
                 cmd.IsShowingInformation = true;
+            }
+
+            Console.WriteLine(GetHelpText(commandName));
+        }
+
+        public string GetHelpText(string commandName = null)
+        {
+            var sb = new StringBuilder();
```

Comment:
Yeah, probably doesn't make much difference. Since perf is not really a concern, I'm fine with whatever is more readable.